### PR TITLE
Feature/test config reforged

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Diverses amelioration of test/lint configuration.

Lint (_in this PR_) is currently broken, I'm waiting your call @vorpaljs-reforged/vorpal-developers,
about the lint configuration, (mostly prettier). Before to auto+manually fix it.

:)

### CODE REVIEW

**Review by hongaar — APPROVED** [review]
🍻 to a sane set of eslint base rules and 🍻 for running prettier through eslint and 🍻 for less scripts scripts and dependencies.

Would recommend however we use all prettier defaults instead of fine-tuning. In my experience, this leads to less friction and avoids bikeshedding.

But easy to change later so 👍 for now.

**Review by Ore4444 — COMMENTED** [review]
i agree with everything, just make sure to rebase to develop before merge

**Comment by AdrieanKhisbe** [comment]
@Ore4444 Rebase made.

Many lint warning remains, mainly about typing.

I might have broken some test, I'll fix it tomorrow morning

**Comment by AdrieanKhisbe** [comment]
@hongaar @Ore4444 and green it is!!! 🎉

**Review by hongaar — COMMENTED** [review]
Awesome, code looks much better.

While I still stand by my original comment about prettier, how about we just revert `tabWidth: 4` to leave it at the default (2) to better match the style of the original vorpal repo. Might make transitioning back upstream easier.

**Comment by AdrieanKhisbe** [comment]
@hongaar as you wish. In fact the codebase state when I selected tabWith was a mix of 2 and 4.

I personnaly feel more comfortable with 4, but I'm more than okey with 2.
We just have to make the call and I'll apply.

cc @vorpaljs-reforged/vorpal-developers

**Comment by AdrieanKhisbe** [comment]
I made a poll on the [slack 🗳 ](https://vorpal-workspace.slack.com/archives/CGE8VN2KC/p1572361235000900)

File: package.json [hunk 1]
```diff
@@ -58,9 +53,6 @@
     "wrap-ansi": "^6.0.0"
   },
   "devDependencies": {
-    "@babel/core": "^7.6.4",
```
Comment (Ore4444) [inline package.json:61]: I do think we should compile the code with Babel. preset-env has some sensible defaults that make the code more compatible. Also, I'm all up for upgrading our syntax with draft proposals

File: src/autocomplete.ts [hunk 2]
```diff
@@ -1,140 +1,143 @@
...
+                    input.context = dataMatch;
+                    evaluateTabs(input);
+                    end(assembleInput(input));
+                } else {
+                    end(filterData(input.context as string, data));
+                }
+            });
         } else {
-          end(filterData(input.context as string, data));
+            end(filterData(input.context as string, commands));
```
Comment (Ore4444) [inline src/autocomplete.ts]: why this?

File: src/session.ts [hunk 3]
```diff
@@ -12,479 +12,475 @@ import util from './util';
...
-          stack = err.stack;
+        if (keyMatch) {
+            if (['up', 'down'].indexOf(key) > -1) {
+                cb(undefined, this.getHistory(key));
+            } else if (key === 'tab') {
+                this.getAutocomplete(value, cb);
+            }
         } else {
-          stack = err;
+            this._histCtr = 0;
```
Comment (Ore4444) [inline src/session.ts]: why this?

**Comment by hongaar** [comment]
If @Ore4444 agrees, I'd say let's merge this as it touches a lot of files (and then we can merge/rebase our branches), and change to another `tabWidth` if needed later (once poll closes).

:shipit: 

I'll continue adding types tomorrow, first 🛌💤

**Comment by AdrieanKhisbe** [comment]
@hongaar I prefer we settle once and for all the formatting.
I don't want to have another changeallfilecommits, these are meant to be exceptionnal.

I change the tabWith to 2, and we go forward :)
(I bet you'll accept even more, than so far I'm the only voice that prefer 4)

**Review by hongaar — APPROVED** [review]

**Review by Ore4444 — APPROVED** [review]

File: src/autocomplete.ts (hunk 2 above)
Comment (AdrieanKhisbe) [reply src/autocomplete.ts]: it seems the feeback is on an outdated part of code. (github flag as outdated, and I dont find this in the diff. (section not modified)

File: src/session.ts (hunk 3 above)
Comment (AdrieanKhisbe) [reply src/session.ts]: idem. i dont know where this diff comes from

**Pull request merged**